pkg/slo-controller/noderesource/plugins/midresource/plugin.go

```diff
@@ -177,39 +170,11 @@ func (p *Plugin) calculate(strategy *configuration.ColocationStrategy, node *cor
 	}
 
 	nodeAllocatable := node.Status.Allocatable
-	defaultStrategy := sloconfig.DefaultColocationStrategy()
-	cpuThresholdRatio := getPercentFromStrategy(strategy, &defaultStrategy, MidCPUThreshold)
-	if maxMilliCPU := float64(nodeAllocatable.Cpu().MilliValue()) * cpuThresholdRatio; allocatableMilliCPU > int64(maxMilliCPU) {
-		allocatableMilliCPU = int64(maxMilliCPU)
-	}
-	if allocatableMilliCPU < 0 {
-		klog.V(5).Infof("mid allocatable cpu of node %s is %v less than zero, set to zero",
-			node.Name, allocatableMilliCPU)
-		allocatableMilliCPU = 0
-	}
-	cpuInMilliCores := resource.NewQuantity(allocatableMilliCPU, resource.DecimalSI)
 
-	memThresholdRatio := getPercentFromStrategy(strategy, &defaultStrategy, MidMemoryThreshold)
-	if maxMemory := float64(nodeAllocatable.Memory().Value()) * memThresholdRatio; allocatableMemory > int64(maxMemory) {
-		allocatableMemory = int64(maxMemory)
-	}
-	if allocatableMemory < 0 {
-		klog.V(5).Infof("mid allocatable memory of node %s is %v less than zero, set to zero",
-			node.Name, allocatableMemory)
-		allocatableMemory = 0
-	}
-	memory := resource.NewQuantity(allocatableMemory, resource.BinarySI)
-
-	// add unallocated
+	// TODO: consider SafetyMargin and NodeReserved
 	unallocated := p.getUnallocated(node, podList)
-	// CPU need turn into milli value
-	unallocatedMilliCPU, unallocatedMemory := resource.NewQuantity(unallocated.Cpu().MilliValue(), resource.DecimalSI), unallocated.Memory()
-	midUnallocatedRatio := getPercentFromStrategy(strategy, &defaultStrategy, MidUnallocatedPercent)
-	adjustedUnallocatedCPU := resource.NewQuantity(int64(float64(unallocatedMilliCPU.Value())*midUnallocatedRatio), resource.DecimalSI)
-	adjustedUnallocatedMemory := resource.NewQuantity(int64(float64(unallocatedMemory.Value())*midUnallocatedRatio), resource.BinarySI)
 
-	cpuInMilliCores.Add(*adjustedUnallocatedCPU)
-	memory.Add(*adjustedUnallocatedMemory)
+	cpuInMilliCores, memory, cpuMsg, memMsg := resutil.CalculateMidResourceByPolicy(strategy, nodeAllocatable, unallocated, allocatableMilliCPU, allocatableMemory, prodReclaimableCPU, prodReclaimableMemory, node.Name)
```

Comment:
"prodReclaimableCPU", is it more appropriate to name it "prodReclaimableMilliCPU" to avoid misused?

Reply:
fix all related problem. @saintube